File: loopr-ios/Models/Token/TokenInfo.swift
//
//  Depth.swift
//  loopr-ios
//
//  Created by kenshin on 2018/3/6.
//  Copyright © 2018年 Loopring. All rights reserved.
//

import Foundation

class TokenInfo {
    let symbol: String
    let circulatingSupply: Int
    let totalSupply: Int
    let maxSupply: Int
    let cmcRank: Int
    let icoRateWithEth: Double
    let websiteURL: String

    init(json: JSON) {
        self.symbol = json["symbol"].stringValue
        self.circulatingSupply = json["circulatingSupply"].intValue
        self.totalSupply = json["totalSupply"].intValue
        self.maxSupply = json["maxSupply"].intValue
        self.cmcRank = json["cmcRank"].boolValue
        self.icoRateWithEth = json["icoRateWithEth"].doubleValue
        self.websiteUrl = json["websiteUrl"].stringValue
    }
}

File: walletsdk/src/main/java/leaf/prod/walletsdk/model/token/TokenInfo.java
/**
 * Created with IntelliJ IDEA.
 * User: kenshin wangchen@loopring.org
 * Time: 2019-03-21 2:32 PM
 * Cooperation: loopring.org 路印协议基金会
 */
package leaf.prod.walletsdk.model.token;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class TokenInfo {

    private String symbol;

    private Long circulatingSupply;

    private Long totalSupply;

    private Long maxSupply;

    private Integer cmcRank;

    private Double icoRateWithEth;

    private String websiteUrl;
}
